core-bundle/src/Resources/contao/classes/DataContainer.php

```diff
@@ -1697,4 +1706,54 @@ public function generateRecordLabel(array $row, string $table = null, bool $prot
 
 		return $label;
 	}
+
+	/**
+	 * @param array<int|string> $ids
+	 */
+	protected function preloadCurrentRecords(array $ids, string $table): void
+	{
+		if (\count($ids) === 1)
+		{
+			$rows = array($this->Database->prepare("SELECT * FROM " . $table . " WHERE id=?")
+				->limit(1)
+				->execute($ids[0])
```

Comment:
If we can' be sure the key is numeric

```suggestion
				->execute(array_shift($ids))
```